homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +247,20 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def _numeric_state_expected(self) -> bool:
+        """Return true if the sensor must be numeric."""
+        if (
+            self.state_class is not None
+            or self.native_unit_of_measurement is not None
+            or self.native_precision is not None
+        ):
+            return True
+        # Sensors with custom device classes are not considered numeric
+        device_class = try_parse_enum(SensorDeviceClass, self.device_class)
+        return not (device_class is None or device_class in NON_NUMERIC_DEVICE_CLASSES)
```

Comment:
I'd unpack the parenthesis and rewrite the condition to make it easier to read.

```py
return device_class is not None and device_class not in NON_NUMERIC_DEVICE_CLASSES
```

Reply:
But then it fits on three lines instead of one

Reply:
I don't think we should make it harder to read the code to make it fit on less lines.

Reply:
https://github.com/home-assistant/core/pull/87111